src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -132,7 +136,65 @@ private async Task GetProjectMetadataAsync(
                     {
                         if (listPackageArgs.ReportType != ReportType.Default)  // generic list package is offline -- no server lookups
                         {
-                            WarnForHttpSources(listPackageArgs, projectModel);
+                            WarnForHttpSources(listPackageArgs, projectModel, auditSources);
+
```

Comment:
The current method `GetProjectMetadataAsync` is about 80 lines. 

It does things like:
- Figure out if the project is compatible
- Get the assets file
- gets the list of requested/resolver eversions and creates a model.
- If need be, populate that model with "online" data in 2 lines.

```cs
                            var metadata = await GetPackageMetadataAsync(frameworks, listPackageArgs);
                            await UpdatePackagesWithSourceMetadata(frameworks, metadata, listPackageArgs);
```

Currently, you're adding about 60 extra lines doing the same thing that the "old path" did in 2. 
That reduces the readability of the method. 

Consider refactoring into smaller methods with appropriate naming.

Reply:
I agree, it looks like lots of opportunity to refactor for reusability.